src/swap/utils/getTransactionType.ts

```diff
@@ -0,0 +1,21 @@
+import type { Call } from '../../transaction/types';
+
+export function getTransactionType(
+  transactions: Call[],
+  index: number,
+): 'ERC20' | 'Permit2' | 'Swap' | null {
```

Comment:
We are using this type in a few places, maybe we want to abstract this.

Reply:
added shared type